bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,219 @@
+                try {
+                    yioremoteDockHandlerwebSocketClient.connect(yioremoteDockwebSocketClient,
+                            uriyiodockwebsocketaddress, yioremoteDockwebSocketClientrequest);
+                    yioremoteDockwebSocketClient.getLatch().await();
+                    Thread.sleep(1000);
+                    try {
+                        if (yioremoteDockwebSocketClient.getbooleanauthenticationrequired()) {
+                            yioremoteDockwebSocketClient.sendMessage(YIOREMOTEMESSAGETYPE.AUTHENTICATE,
+                                    localConfig.accesstoken);
+                            Thread.sleep(1000);
+                            if (yioremoteDockwebSocketClient.getbooleanauthenticationok()) {
+                                yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATED;
+                                logger.debug("authentication to YIO dock ok");
+                                updateStatus(ThingStatus.ONLINE);
+                                try {
+                                    Runnable heartbeatpolling = new Runnable() {
+                                        @Override
+                                        public void run() {
+                                            try {
+                                                if (yioremotedockactualstatus
+                                                        .equals(YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATED)) {
+                                                    yioremoteDockwebSocketClient
+                                                            .sendMessage(YIOREMOTEMESSAGETYPE.HEARTBEAT, "");
+
+                                                    Thread.sleep(1000);
+
+                                                    if (yioremoteDockwebSocketClient.getbooleanheartbeat()) {
+                                                        logger.debug("heartbeat ok");
+                                                        triggerChannel(getChannelUuid(GROUP_OUTPUT, YIODOCKSTATUS));
+                                                        updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS,
+                                                                yioremoteDockwebSocketClient.getstringreceivedstatus());
+                                                    } else {
+                                                        yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.CONNECTIONFAILED;
+                                                        updateStatus(ThingStatus.OFFLINE,
+                                                                ThingStatusDetail.CONFIGURATION_ERROR,
+                                                                "Connection lost no ping from YIO DOCK");
+                                                        updateState(GROUP_OUTPUT, YIODOCKSTATUS, UnDefType.UNDEF);
+                                                        pollingJob.cancel(true);
```

Comment:
As this is nullable, you need to do a null check before. You can store `pollingJob` to a local variable and do the null check on that.

Reply:
this is what i don't understand exactly can you explain it a bit more

Reply:
`pollingJob` can be null at this point, because it isn't inittialized in the constructor. So, you need to check if it is not null before accessing it: `if (pollingJob != null)`.

But this would result in a compiler warning, as `pollingJob` is a field and could theoretically be modified by another thread exactly between the null check and the access in the next line.

To mitigate this, you can store `pollingJob` to a local variable before checking it:

```
                                                        Future<?> localPollingJob = pollingJob;
                                                        if (localPollingJob != null) {
                                                            localPollingJob.cancel(true);
                                                        }
```

Reply:
Resolved